shared/rollup.config.mjs

```diff
@@ -31,7 +32,7 @@ export function createConfig({ pkg, external = [] }) {
       {
         format: 'es',
         file: pkg.module,
-        plugins: [emitModulePackageFile()],
+        plugins: [emitModulePackageFile(), emitDeclarationFile()],
```

Comment:
to fix the CJS types you could just emit this for CJS as well + just remove the `types` condition from all `package.json` files. If the layout would always look like this then it would just work:
```
lib/cjs/index.js
lib/cjs/index.d.ts
lib/es/index.js
lib/es/index.d.ts
lib/es/package.json#type:module
```

That assumes that the declarations are self-contained and that they can safely be copied without breaking any relative imports etc in them.

Reply:
This solution looks cleaner, I tested this modification locally and it works fine